### DISCUSSION TITLE
How do I set canonical for blog list pages?

### DISCUSSION THREAD

**Discussion by pranay01** [facebook/docusaurus#5924] [Q&A]
I have multiple blogs running in the same site - 
https://signoz.io/opentelemetry/
https://signoz.io/learn/

The canonical for these pages seems to be set randomly ( or the very first blog hosted in the corresponding folder?) 
How can I manually specify canonical URLs for these blog list pages? Since there is no .md or .js file I am using to create these pages, I can't use the `<Head>` tag

**Comment by slorber** [comment]
Hi,

It's not clear to me why you have these canonical URLs on blog list pages.

I couldn't find your site sourcecode

Make sure you are using the latest Docusaurus version and that it's not related to any swizzled blog component (in which case you'd. need to update them).

**Reply by pranay01** [reply]
the point is that currently the canonical URL for these pages is randomly selected, which sometimes points to old blog post, and google doesn't. like it ( gives 404 etc)

If we can set the canonical URLs manually for blog list pages also, that would be great. I  on docusaurus beta8

**Reply by slorber** [reply]
It's not the case on the Docusaurus blogs:
 - https://docusaurus.io/blog
 - https://docusaurus.io/tests/blog

That's why I suggest you to double-check your site and in particular swizzled components, that you are responsible to maintain and keep up to date. Unless you show me your site's source code, I can't help you nor trust you that this is a Docusaurus problem, but rather trust that it is a problem specific to your site.

> If we can set the canonical URLs manually for blog list pages also, that would be great. I on docusaurus beta8

By default canonical URLs are set to the same URLs as blog list page URLs. If you want to do something else (why?), then you'd have to use swizzle
